### PR TITLE
[test]:  fixed failing test & add delete patient test

### PR DESCRIPTION
**Pull request by valens200**
Fixed the failing test which was failing all the times.

Initially, this test expects to receive several patients equal to one but in real implementation, the output of `"patientService.getAllPatients().size()"` is zero which leads to the failure of the test. 

I also added delete patient test to ensure the high quality of the functionality.

### CODE REVIEW

**Comment by valens200** [comment]
@mozzy11  @reagan-meant you can review this PR and suggest changes. 
Thank you.

File: volume/tomcat/hapi_server.xml [hunk 1]
```diff
@@ -19,17 +19,17 @@
      define subcomponents such as "Valves" at this level.
      Documentation at /docs/config/server.html
  -->
-<Server port="-1" shutdown="SHUTDOWN">
```
Comment (mozzy11) [inline volume/tomcat/hapi_server.xml:22]: get rid of these changes , its probably updated the by the container volume.

**Comment by mozzy11** [comment]
Thanks @valens200 , 
did your tests pass locally ??

**Comment by valens200** [comment]
Yes @mozzy11, I think the action build is failing due to changing this port as you requested.
Screen Shot 2024-03-18 at 21 57 53

Setting it to -1 makes the tests to failure because the data source can't be found.

**Comment by valens200** [comment]
> Thanks @valens200 , did your tests pass locally ??



> @mozzy11 @reagan-meant you can review this PR and suggest changes. Thank you.

yes

**Comment by mozzy11** [comment]
@valens200 i dont think thats right , 
These are integration tests at the service layer and they dont have any thing to do with the servlet container .

File: src/test/java/org/openelisglobal/patient/PatientServiceTest.java [hunk 2]
```diff
@@ -35,8 +38,27 @@ public class PatientServiceTest {
...
+		Assert.assertEquals(1, patientService.getAllPatients().size());
+	}
+	@Test
+	public void deletePatient_shouldDeleteNewPatient() throws Exception {
+		String firstName = "Peter";
+		String lastname = "Master";
+		String dob = "9/12/1993";
+		String gender = "M";
+		Patient pat = createPatient(firstName, lastname, dob, gender);
+		String patientId = patientService.insert(pat);
```
Comment (mozzy11) [inline src/test/java/org/openelisglobal/patient/PatientServiceTest.java:55]: can you assert the patient exists before delete ?

File: volume/tomcat/hapi_server.xml (hunk 1 above)
Comment (mozzy11) [reply volume/tomcat/hapi_server.xml:22]: we dont need these changes here for the tests to pass

**Review by mozzy11 — CHANGES_REQUESTED** [review]
Get rid of the server,xml changes and also add an assertion before delete

**Comment by valens200** [comment]
I see @mozzy11 , let me apply the changes

**Comment by valens200** [comment]
> @valens200 i dont think thats right , These are integration tests at the service layer and they dont have any thing to do with the servlet container .

Yes sir @mozzy11

**Comment by valens200** [comment]
@mozzy11 I think that's all.

**Comment by valens200** [comment]
@mozzy11 you can check for changes.

**Review by mozzy11 — CHANGES_REQUESTED** [review]
Thanks @valens200 . 
you still have format changes in hapi_server.xml file . 
amd also can you also put the Guide changes in a separate PR ??

**Comment by valens200** [comment]
@mozzy11 If you are here please you can see re-review this.

**Pull request closed without merging**